### ISSUE TITLE
error in no-useless-constructor

### ISSUE THREAD

**Issue by TrejGun** [eslint/eslint#5426]
**What version of ESLint are you using?**
2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
babel-eslint

**What did you do? Please include the actual source code causing the issue.**
```js
class A {}

class B extends A {
	constructor() {
		super(...arguments);
	}
}
```

**What did you expect to happen?**
eslint will exit with code 0

**What actually happened? Please include the actual, raw output from ESLint.**
```
...eslint/lib/rules/no-useless-constructor.js:33
                ctorParams[index].type === "RestElement" &&
                                 ^

TypeError: Cannot read property 'type' of undefined
    at ...eslint/lib/rules/no-useless-constructor.js:33:34
    at Array.every (native)
    at isRedundantSuperCall (...eslint/lib/rules/no-useless-constructor.js:29:45)
    at EventEmitter.checkForConstructor (...eslint/lib/rules/no-useless-constructor.js:52:46)
    at emitOne (events.js:95:20)
    at EventEmitter.emit (events.js:182:7)
    at NodeEventGenerator.enterNode (...eslint/lib/util/node-event-generator.js:42:22)
    at CodePathAnalyzer.enterNode (...eslint/lib/code-path-analysis/code-path-analyzer.js:591:23)
    at CommentEventGenerator.enterNode (...eslint/lib/util/comment-event-generator.js:98:23)
    at Controller.controller.traverse.enter (...eslint/lib/eslint.js:839:36)
```

**Comment by faceleg** [comment]
I'm not able to replicate this by:

```
cd ~/temp
# create temp.js file with contents pasted from above
npm install eslint babel-eslint --save-dev
# creating an .eslintrc file as describe here https://github.com/babel/babel-eslint#setup
node_modules/.bin/eslint test.js
```

**Comment by faceleg** [comment]
Assigned to self while evaluating.

**Comment by faceleg** [comment]
Possibly related to: https://github.com/eslint/eslint/issues/5355#issuecomment-186814689

Please consider opening an issue on babel/babel-eslint.

**Comment by TrejGun** [comment]
sorry here is `.eslintrc` to reproduce the issue
```.json
{
  "parser": "babel-eslint",
  "rules": {
    "no-useless-constructor": 2
  }
}
```

**Comment by michaelficarra** [comment]
@trejgun Please try to repro using master. This rule has changed very significantly since the last release, and I would be surprised if this bug still exists.

**Comment by alberto** [comment]
Confirmed it no longer throws in master.

**Comment by fixpunkt** [comment]
For the benefit of anybody also coming here from Google: This issue is fixed in 2.3.0.
